### PR TITLE
signup and signin and refresh token functionalities implemented

### PR DESCRIPTION
**Pull request by IslomJonMirzakbarov**

### CODE REVIEW

File: src/components/ControlledFormElements/HFTextField.jsx [hunk 1]
```diff
@@ -20,7 +21,19 @@ const HFTextField = ({
         name={name}
         defaultValue=""
         rules={{
-          required: required ? `Provided ${name} is incorrect` : false,
+          required: required ? `Enter your ${name}` : false,
+          ...(name === 'password' && {
+            minLength: {
+              value: 8,
```
Comment (otabeknos1rov) [inline src/components/ControlledFormElements/HFTextField.jsx:27]: can we make it dynamic, coming from props?

File: src/utils/request.js [hunk 2]
```diff
@@ -1,25 +1,100 @@
...
-import { showAlert } from '../store/alert/alert.thunk';
+import axios from 'axios'
+import { store } from '../store/index'
+import { authActions } from '../store/auth/auth.slice'
+import { showAlert } from '../store/alert/alert.thunk'
 
-export const baseURL = process.env.REACT_APP_BASE_URL;
-console.log(baseURL);
+export const baseURL = process.env.REACT_APP_BASE_URL
+console.log(baseURL)
```
Comment (otabeknos1rov) [inline src/utils/request.js:7]: how it comes to delete all logs?

File: src/views/Auth/Reset/index.jsx [hunk 3]
```diff
@@ -1,14 +1,47 @@
...
+      )
+      setTimeout(() => {
+        setError(null)
+        reset()
+      }, 3000)
+    }
+  }
+
+  useEffect(() => {
+    let timer
```
Comment (otabeknos1rov) [inline src/views/Auth/Reset/index.jsx:35]: why do we need timer here?

File: src/components/ControlledFormElements/HFTextField.jsx (hunk 1 above)
Comment (IslomJonMirzakbarov) [reply src/components/ControlledFormElements/HFTextField.jsx:27]: yes, sure checking...

File: src/utils/request.js (hunk 2 above)
Comment (IslomJonMirzakbarov) [reply src/utils/request.js:7]: yes, working on...

File: src/views/Auth/Reset/index.jsx (hunk 3 above)
Comment (otabeknos1rov) [reply src/views/Auth/Reset/index.jsx:35]: it is not recommended to use useEffect, please do refactor find better solution

File: .env [hunk 4]
```diff
@@ -1 +1 @@
-REACT_APP_BASE_URL=http://3.39.9.237:3000/
\ No newline at end of file
+REACT_APP_BASE_URL=http://3.39.9.237:3000/v1/
```
[the remark this answers is not in the document]
Comment (IslomJonMirzakbarov) [reply .env:1]: backend developer updated the api adding v1

File: src/views/Auth/Reset/index.jsx (hunk 3 above)
Comment (IslomJonMirzakbarov) [reply src/views/Auth/Reset/index.jsx:35]: ok, working on it

**Pull request merged**